### ISSUE TITLE
Allow resize on a single axis with Ctrl+drag

### ISSUE THREAD

**Issue by bstee615** [SVG-Edit/svgedit#346]
**Describe the bug**
I want to be able to scale an item and keep one axis static.

**Expected behavior**
My suggestion is to do how many other editors do and allow the user special controls when they drag a resize handle while holding down the Ctrl key. For example, I could hold down Ctrl, grab the resize handle, and it would drag along the X axis or Y axis (whichever's closest).

SVG-Edit Version: 5.0.0

I may be able to fulfill this request - I haven't taken a look at your source yet, but I'd love to contribute to such a great project.

**Comment by brettz9** [comment]
Sounds like a nice idea to me!

**Comment by bstee615** [comment]
Yes, while I was using this tool I missed the feature from GIMP. In GIMP there is something similar, where resizing a selection scales it from the center. Here's a clip of what GIMP does: 

![2fe4b49d-8495-4e4f-9071-74fc3eba452e](https://user-images.githubusercontent.com/22305849/60343613-00ccea80-997a-11e9-9fa4-3747585b7358.gif)

Personally, I would like the axis-lock feature more as it would be used more frequently. But it's open to discussion. Unless anyone objects, I'm going to go with the axis-lock.

I'll work on adding this and submit a PR.

**Comment by brettz9** [comment]
Excellent! I just ask to assist me in reviewing PRs that you try to make the code well-documented. Thanks!

**Comment by bstee615** [comment]
Hi, how can I build to test the behavior locally? I wrote some unit tests and can run those, but I cannot figure out how to build the project and embed it locally. Thank you.

**Comment by brettz9** [comment]
Did you take a look at https://github.com/SVG-Edit/svgedit/#es6-modules-file ? Which instructions weren't clear. Others have been unclear, and I recently cleaned up the README a little to try to make it more clear. I'd be happy to go over any aspects that are not clear, but would need to know what you've tried and are trying to do.

You can ignore the instructions about creating your own npm package and rather than adding svgedit as a dependency, you can just clone the repository. However, you do need to use `npm` (which you can get from installing Node). `npm` is used to install some development dependencies and run the build routines.

**Comment by bstee615** [comment]
Thank you for referring me to that guide. Maybe you could rename `npm run build-config` to be clearer - as it is I thought it was a command to build some kind of config file rather than to build using a different configuration.

With your tip, I've got it working to test out my changes - I've got a couple logical errors and some code cleanup to do, then it will be ready for PR.

**Comment by brettz9** [comment]
I've renamed the script to `build-by-config` instead.

**Comment by bstee615** [comment]
Thank you, I think the new name is much more accessible.
